tests/30rooms/30history-visibility.pl

```diff
@@ -534,24 +534,27 @@ sub test_history_visibility
       });
    };
 
-test "Backfill works correctly with history visilibty set to joined",
+test "Backfill works correctly with history visibility set to joined",
    requires => [ magic_local_user_and_room_fixtures( with_alias => 1), local_user_fixture(), remote_user_fixture() ],
 
    do => sub {
       my ( $user, $room_id, $room_alias, $another_user, $remote_user, $room_alias_name ) = @_;
 
       matrix_set_room_history_visibility( $user, $room_id, "joined" )
       ->then( sub {
+         # Send some m.room.message that the remote server will not be able to see
          repeat( sub {
             my $msgnum = $_[0];
 
             matrix_send_room_text_message( $user, $room_id, body => "Message $msgnum" );
-         }, foreach => [ 1 .. 100 ]);
+         }, foreach => [ 1 .. 10 ]);
       })->then( sub {
          matrix_join_room( $another_user, $room_alias );
       })->then( sub {
          matrix_send_room_text_message( $user, $room_id, body => "2" );
       })->then( sub {
+         # We now send a state event to ensure they're correctly handled in
```

Comment:
erm; is it not the *other* join that we're doing to test this?

This join is necessary for us to see anything at all?

Reply:
Oh I'm such a crank. Sorry, yes, imagine that comment on the previous join.